src/datasets/arrow_dataset.py

```diff
@@ -3802,12 +3796,48 @@ def sort(
         if len(self) == 0:
             return self
 
-        # Check the column name
-        if not isinstance(column, str) or column not in self._data.column_names:
+        # Deprecation warning
+        if kind != "deprecated":
+            warnings.warn(
+                "The 'kind' parameter is deprecated and will be removed in future releases.",
+                category=DeprecationWarning,
+            )
```

Comment:
We use `FutureWarning` to notify users about deprecations

```suggestion
        if kind != "deprecated":
            warnings.warn(
                "`kind` was deprecated in version 2.10.0 and will be removed in 3.0.0.",
                category=FutureWarning,
            )
```